src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -1251,6 +1281,18 @@ def test_process_not_supported_declarations():
     l11sym = fake_parent.symbol_table.lookup("l11")
     assert isinstance(l11sym.datatype, UnknownFortranType)
 
+    # Assumed-size array with specified upper bound. fparser2 does spot that
+    # this is invalid so we have to break the parse tree it produces for
+    # a valid case.
+    reader = FortranStringReader("integer :: l12(:)")
+    fparser2spec = Fortran2003.Specification_Part(reader).content[0]
+    # Break the parse tree
+    aspec = walk(fparser2spec, Fortran2003.Assumed_Shape_Spec)[0]
+    aspec.items = (None, Fortran2003.Int_Literal_Constant('4', None))
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    l11sym = fake_parent.symbol_table.lookup("l12")
+    assert isinstance(l11sym.datatype, UnknownFortranType)
```

Comment:
Is this equivalent to "integer :: l12(:4)" ? In this case gfortran does not accept it. Is it valid Fortran? Should it be a GenerationError instead.

Reply:
No, it's not valid Fortran and yes, that's a good idea - I've changed it to a GenerationError now.